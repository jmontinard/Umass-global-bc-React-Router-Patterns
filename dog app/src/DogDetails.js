import React from 'react';
import { useParams, Link } from 'react-router-dom';


const DogDetails = ({ dogs }) => {
  const { name } = useParams();
  const currentDog = dogs.find(dog => dog.name.toLowerCase() === name.toLowerCase());

  if (!currentDog) {
    return <div className="dog-details">Dog not found</div>;
  }

  return (
    <div className="dog-details">
      <h1>{currentDog.name}</h1>
      <img src={currentDog.src} alt={currentDog.name} />
      <p>Age: {currentDog.age}</p>
      <ul>
        {currentDog.facts.map((fact, index) => (
          <li key={index}>{fact}</li>
        ))}
      </ul>
      <Link to="/dogs">Back to Dog List</Link>
    </div>
  );
};

export default DogDetails;
