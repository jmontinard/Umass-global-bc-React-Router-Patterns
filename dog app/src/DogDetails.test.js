// DogDetails.test.js
import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import DogDetails from './DogDetails';

test('renders DogDetails component without crashing', () => {
  const dogs = [
    {
      name: 'Whiskey',
      age: 5,
      src: 'whiskey.jpg',
      facts: ['Whiskey loves eating popcorn.'],
    },
  ];

  render(
    <MemoryRouter initialEntries={['/dogs/Whiskey']}>
      <Routes>
        <Route
          path="/dogs/:name"
          element={<DogDetails dogs={dogs} />}
        />
      </Routes>
    </MemoryRouter>
  );
});

test('matches snapshot for DogDetails component', () => {
  const dogs = [
    {
      name: 'Whiskey',
      age: 5,
      src: 'whiskey.jpg',
      facts: ['Whiskey loves eating popcorn.'],
    },
  ];

  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/Whiskey']}>
      <Routes>
        <Route
          path="/dogs/:name"
          element={<DogDetails dogs={dogs} />}
        />
      </Routes>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
