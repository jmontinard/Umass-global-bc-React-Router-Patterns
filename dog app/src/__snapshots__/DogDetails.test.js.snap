// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for DogDetails component 1`] = `
<div>
  <div
    class="dog-details"
  >
    <h1>
      Whiskey
    </h1>
    <img
      alt="Whiskey"
      src="whiskey.jpg"
    />
    <p>
      Age: 
      5
    </p>
    <ul>
      <li>
        Whiskey loves eating popcorn.
      </li>
    </ul>
    <a
      href="/dogs"
    >
      Back to Dog List
    </a>
  </div>
</div>
`;
